docs/pages/blog/making-customizable-components.md

```diff
@@ -0,0 +1,287 @@
+ opacity: 1;
+}
+```
+
+### Let JS generate the CSS
+
+Maybe you don’t want to spent your time switching between CSS and JS files, or write long and cluttered stylesheets.
+To avoid these problems you can integrate the style in your JS code 🎉
+
+Because the level of customization varies across projects, MUI has added multiple ways to customize components.
+If you want more information on this topic, you can have a look at the [documentation](https://mui.com/material-ui/customization/how-to-customize/).
+
+## Logic modification
+
+Styling is not the only thing you need to customize.
+You may have to modify the logic, which will have to be handled by the component itself.
+
+### Add a prop
+
+Consider a component that lets users rate a product.
+Going from 1 to 5 stars is good.
+But maybe you need to go up to 10.
+To do so we can add a `max` prop and render as many stars as the value of `max`.
+
+That works well enough for a simple UI element, but what happens when the component in question has many more moving parts?
+
+
+### It’s never that simple
+
+Let’s play with a slightly more complex component: the [`DataGrid`](https://mui.com/x/react-data-grid/).
+This component allows you to manage data by applying sorting, filtering, editing, exporting, and many other -INGs.
+
+Consider the following feature request: And together we will try to answer an open issue I get.
+
+:::info
+[DataGrid] Sorting column options by alphabetical order
+
+When I open the filterPanel, the input listing the names of the columns is sorted according to column position.
+I would like to sort it by alphabetical order.
+<img src="/static/blog/making-customizable-components/issueScreenshot.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Screen shot of the filter panel with column selector un sorted" />
+:::
+
+This request makes sense.
+When you have a lot of columns, sorting them can make it easier to browse the list.
+But how should we implement this kind of customization?
+
+Adding a prop called `filterPanelColumnInputSortingStrategy` could work, but please don’t do that.
+It just doesn’t scale.
+There are too many different properties that developers might need to modify.
+You will end up with API documentation so long that it will take an eternity to scroll to the end—so nobody will read it.
+
+<img src="/static/blog/making-customizable-components/bruce.gif" style="width: 500px; margin-top: 16px; margin-bottom: 8px;" alt="Your user opening the list of props" />
+
+Here are a few better solutions that can scale more efficiently with complex components.
+
+### Other solutions
+
+#### Don’t bother with components
+
+Passing all the parameters as props of a single component does not work.
+So just don’t create components.
+
+It’s not a joke—that’s the approach of headless libraries such as react-table.
+Instead of providing working components, they only provide hooks for managing the features, and let developers build their components on top of that.
```

Comment:
To make the terminology more neutral (only & that).

```suggestion
Instead of providing working components, they provide hooks for managing the features and let developers build their components on top of it.
```